javascript/ql/src/semmle/javascript/GlobalAccessPaths.qll

```diff
@@ -412,4 +412,16 @@ module AccessPath {
       isAssignedInUniqueFile(name)
     )
   }
+
+  /**
+   * Gets a SourceNode that is accessed using the same access path as the input. 
+   */
+  DataFlow::SourceNode getASourceAccess(DataFlow::Node node) {
```

Comment:
How about `getAnAliasedSourceNode`? We already use the terms "source" and "access" quite liberally so the name "source access" becomes a little vague.

Reply:
Also, I'm tempted to say there should be a `pragma[inline]` on here as this is too large to materialize. Note that this includes global access paths, not just local ones, which is why the all-pairs access blows up very easily.

It worries me when we have  predicates that have an unfeasible size as-written, as the cost may be temporarily hidden by an optimization, which then stops working at some point in the future.